### ISSUE TITLE
Oracle Database Server 12.1.0.2 Docker Image 

### ISSUE THREAD

**Issue by arunachg** [openshift/origin#16272]
Hi,
I have created the container in openshift for the oracle database and exposed the service port as NodePort. But a not able to connect from outside.

Is there a specific configuration to use here? 

Anyone got succeeded in deploying oracle db in openshift?

##### Version
oc v1.5.0+031cbe4
kubernetes v1.5.2+43a9be4
features: Basic-Auth GSSAPI Kerberos SPNEGO

Server https://xxxxxx:8443
openshift v3.5.5.15
kubernetes v1.5.2+43a9be4

**Comment by arunachg** [comment]
Find the deployment template below for the OC deployment,

projectapiVersion: v1
apiVersion: v1
kind: Service
metadata:
  labels:
    app: database-enterprise
  name: "database-enterprise"
  namespace: "fre-int"
spec:
  ports:
  - name: 1521-tcp
    port: 1521
    nodePort: 31521
  - name: 5500-tcp
    port: 5500
    nodePort: 31522
  selector:
    app: database-enterprise
  type: NodePort
---
apiVersion: v1
kind: Route
metadata:
  labels:
    app: database-enterprise
  name: "database-enterprise"
  namespace: "fre-int"
spec:
  host:
  port:
    targetPort: 5500-tcp
  to:
    kind: Service
    name: "database-enterprise"
    weight: 100
---
apiVersion: v1
kind: DeploymentConfig
metadata:
  labels:
    app: database-enterprise
  name: database-enterprise
  namespace: "fre-int"
spec:
  replicas: 1
  selector:
    app: database-enterprise
    deploymentconfig: database-enterprise
  strategy:
    type: Rolling
  template:
    metadata:
      labels:
        app: database-enterprise
        deploymentconfig: database-enterprise
    spec:
      containers:
      - image: database-enterprise:latest
        imagePullPolicy: IfNotPresent
        name: database-enterprise
        ports:
        - containerPort: "1521"
          protocol: TCP
        - containerPort: "5500"
          protocol: TCP
        terminationMessagePath: /dev/termination-log
      restartPolicy: Always
      securityContext: {}
      terminationGracePeriodSeconds: 30
  triggers:
  - type: ConfigChange
  - imageChangeParams:
      automatic: true
      containerNames:
      - database-enterprise
      from:
        kind: ImageStreamTag
        name: database-enterprise:latest
        namespace: "fre-int"
    type: ImageChange

**Comment by arunachg** [comment]
I have exposed the service port through NodePort. The Oracle Database image Which i have used is official one.

**Comment by arunachg** [comment]
@mfojtik any luck from your side. Have you tried to run the DB image in OC

**Comment by nouhouari** [comment]
I have issues when deploying the official images (+slim) to OpenShift. 
Permission denied /home/oracle/bin/initDocker.sh

**Comment by nouhouari** [comment]
You can use 
`oc port-forward <pod> 1521`
to forward the pod port locally. You should be able to connect to localhost:1521.

But, using the official image can't work. Openshift will not run as **oracle** user.
It will use a root user based on project users range.
I've done some changes on the official image and update the scripts to be able to run.

**Comment by daniyalj** [comment]
@arunachg  I know this a while ago, but any luck in getting access externally? I've got the oracle container up and running on openshift but Im unable to hit <exposed-service-FQDN>/em/ (exposed on port 5500)

**Comment by yves-vogl** [comment]
I guess you are running into a privilege problem. We are experiencing a similar issue (ORA-01031). The only workaround is to allow running the container with root privilegies (anyuid). But this does not work on OpenShift online and is a security concern anyway.

**Comment by hichem-belhocine** [comment]
I need to deploy Oracle docker image 12.1.0.2-ee on OpenShift using a YAML file

On GKE works fine.
On Docker also works fine (docker run ....)

But on OpenShift, I always get errors. I know that the security on Openshift is more complicated but i have done all the necessary steps to make it run.

I have created a service account, then assign a Security Context Constraints anyuid to it, but it doesn't work.

I have tested this example: https://medium.com/@pittar/running-oracle-12c-on-openshift-container-platform-ca471a9f7057
It works fine, but i need to run the 12.1.0.2-ee version.

[logs.txt](https://github.com/openshift/origin/files/4640282/logs.txt)
[oracle-deployment.txt](https://github.com/openshift/origin/files/4640285/oracle-deployment.txt)
